crates/node/p2p/src/net/driver.rs

```diff
@@ -235,7 +232,7 @@ impl Network {
                             warn!(err = ?send_err, "Impossible to send a request to the discovery handler. The channel connection is dropped.");
                         }
                     },
-                    req = rpc.recv() => {
+                    req = rpc.as_mut().unwrap().recv(), if rpc.is_some() => {
```

Comment:
I think we can use an [`OptionFuture`](https://docs.rs/futures/latest/futures/future/struct.OptionFuture.html) here instead of unwrapping.